2-Draft-Accepted/RFCxxxx-Secrets-Management.md

```diff
@@ -0,0 +1,114 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Security
+Comments Due: July 31st, 2019
+Plan to implement: Yes, PS7
+---
+
+# Simplify and secure use of secrets in scripts
+
+Advanced scripts that touch many systems require multiple secrets and
+types of secrets particularly when orchestrating across different clouds.
+Best practice is to not hard code any secrets in scripts, but currently
+this requires custom code on different platforms to handle this securely.
+
+This RFC proposes new cmdlets to make this simpler and secure.
+
+## Motivation
+
+> As a PowerShell script author,
+> I can securely use multiple secrets,
+> so that I can automate complex orchestration across multiple remote resources.
+
+## User Experience
+
+Registering secrets for later use:
+
+```powershell
+# Install the Az.KeyVault module to be used in this example.
+# This module doesn't conform to the extensions model so would need to be
+# updated to actually work.
+Install-Module Az.KeyVault
+
+# In this example, we explicitly register this extension
+Register-SecretsVaultExtension -Name Azure -Cmdlet Get-AzKeyVaultSecret -Module Az.KeyVault -Version 1.0.0
+
+# When using remote vaults, expectation is that secrets are
+# stored using their tooling, so this module is only for retrieving secrets
+# from remote vaults
+$azDevOpsToken = Get-Secret -Vault Azure -Name AzDevOps
+Invoke-RestMethod https://dev.azure.com/... -Credential $azDevOpsToken -Authentication Basic
+
+# For local vault, we can register custom secrets
+# In this example, we store a PSCredential object
+Register-Secret -Secret $cred -Name MyCreds
```

Comment:
Should the cred be used here, when we retrieve it or both?

- here is convient
- when you retrieve it is more secure.

Reply:
Should this be
```suggestion
Register-SecretVault -Secret $cred -Name MyCreds
```